File: dotnet/ReportGenerator-VB/Tests/HtmlReportTest.vb
Imports System.Collections.Generic
Imports System.IO
Imports NUnit.Framework

Namespace ReportGeneratorVB.Tests
    <TestFixture()> _
    Public Class HtmlReportTest

        <Test()> _
        Public Sub testReport()
            Dim line As IList(Of Machine) = New List(Of Machine)()
            line.Add(New Machine("mixer", "left"))

            Dim extruder As New Machine("extruder", "center")
            extruder.Put("paste")
            line.Add(extruder)

            Dim oven As New Machine("oven", "right")
            oven.Put("chips")
            line.Add(oven)

            Dim robot As New Robot()
            robot.MoveTo(extruder)
            robot.Pick()

            Dim output As New StringWriter()
            HtmlReport.report(output, line, robot)

            Dim expected As String = "<h1>FACTORY REPORT</h1>" & vbLf + "<ul>" & vbLf & "<li>Machine mixer</li>" & vbLf & "<li>Machine extruder</li>" & vbLf + "<li>Machine oven bin=chips</li>" & vbLf & "</ul>" & vbLf + "<p>Robot location=extruder bin=paste</p>" & vbLf + "<hr>" & vbLf

            Assert.That(expected, [Is].EqualTo(output.ToString()))
        End Sub
    End Class
End Namespace

File: dotnet/ReportGenerator-CS/Tests/HtmlReportTest.cs
using System.Collections.Generic;
using System.IO;
using NUnit.Framework;

namespace ReportGeneratorCS.Tests
{
    [TestFixture]
    public class HtmlReportTest {

        [Test]
        public void testReport() {
            IList<Machine> line = new List<Machine>();
            line.Add(new Machine("mixer", "left"));

            Machine extruder = new Machine("extruder", "center");
            extruder.Put("paste");
            line.Add(extruder);

            Machine oven = new Machine("oven", "right");
            oven.Put("chips");
            line.Add(oven);

            Robot robot = new Robot();
            robot.MoveTo(extruder);
            robot.Pick();

            StringWriter output = new StringWriter();
            HtmlReport.report(output, line, robot);

            string expected = "<h1>FACTORY REPORT</h1>\n"
                              + "<ul>\n<li>Machine mixer</li>\n<li>Machine extruder</li>\n"
                              + "<li>Machine oven bin=chips</li>\n</ul>\n"
                              + "<p>Robot location=extruder bin=paste</p>\n" + "<hr>\n";

            Assert.That(expected, Is.EqualTo(output.ToString()));
        }
    }
}
